### DISCUSSION TITLE
Is there a way to pre-fill the ref and inputs for a workflow_dispatch via query string?

### DISCUSSION THREAD

**Discussion by the-avid-engineer** [community/community#51159] [Actions]
### Select Topic Area

Product Feedback

### Body

Given that a repository has a workflow with an `on: workflow_dispatch` trigger, and possibly inputs, is it possible to create a link to the workflow such that if you click on `Run workflow`, the ref and inputs are pre-filled? I can't find anything like this, but maybe it's an undocumented feature? :joy:

Something like `/actions/workflows/CD.yml?tag=2.0.0&inputs={"environment":"production"}` (URL-encoded, of course) would present the pop-up like this:

<img width="344" alt="image" src="https://user-images.githubusercontent.com/36554739/228044043-32b30079-e8bb-4e36-90c5-d5e55d751a7b.png">

If not, is this the right place to request such a feature?

My immediate thoughts are that:
- The feature should be opt-in and per-workflow
- Maybe you can configure a warning to be displayed when ref/inputs are pre-filled

---

My use-case is a rollback feature. Up to this point, my team's CI/CD pipeline has been just recording a link to a specific workflow run (e.g., `/actions/runs/<run-id>`); when we want to roll back, we just open the link and click `Re-run all jobs`.

![image](https://user-images.githubusercontent.com/36554739/228045458-9425380c-6ecf-4f49-8a31-c4b736e63c33.png)

This works for a limited amount of time, but eventually the `Re-run all jobs` button goes away. We're not sure if it's time-based or something else, but we have noticed it as a short-coming of our rollback procedure.

![image](https://user-images.githubusercontent.com/36554739/228045331-5c7713e0-454c-4604-b03d-6b9e4cfb1100.png)

(Edit: our current pipeline does not use workflow_dispatch, but we are considering a re-worked pipeline that would use workflow_dispatch, and it would be nice to address the issue)

**Comment by JiangHongTiao** [comment]
We would benefit from this as well, as we would like to automate release version numbers to decrease possibility of error.

**Comment by JoshuaSoileau** [comment]
+1 to this as a feature, it'd be real nice to pass URL parameters to the action page as a way to pre-fill input values.

**Comment by faber-yx** [comment]
+1 would be really useful for me too...

**Comment by aschiffmann** [comment]
+1 for urls leading to prefilled workflow_dispatch pages!

**Comment by Niek** [comment]
`?inputs[name]=val` would be nice to have - it's a one-liner in JS to add 🙂

**Comment by emanueleaina** [comment]
Similar requests:

* https://github.com/orgs/community/discussions/28505
* https://github.com/orgs/community/discussions/168896
* https://github.com/orgs/community/discussions/58253
